test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -597,43 +597,40 @@ TEST_F(SingleBalancerTest, InitiallyEmptyServerlist) {
 TEST_F(SingleBalancerTest, Fallback) {
   const int kFallbackTimeoutMs = 200 * grpc_test_slowdown_factor();
   const int kServerlistDelayMs = 500 * grpc_test_slowdown_factor();
-  const int kCallDeadlineMs = 1000 * grpc_test_slowdown_factor();
+  const size_t kNumBackendInResolution = backends_.size() / 2;
 
   ResetStub(kFallbackTimeoutMs);
   std::vector<AddressData> addresses;
   addresses.emplace_back(AddressData{balancer_servers_[0].port_, true, ""});
-  addresses.emplace_back(AddressData{balancer_servers_[1].port_, true, ""});
-  addresses.emplace_back(AddressData{balancer_servers_[2].port_, false, ""});
+  for (size_t i = 0; i < kNumBackendInResolution; ++i) {
+    addresses.emplace_back(AddressData{backend_servers_[i].port_, false, ""});
+  }
   SetNextResolution(addresses);
 
-  // First response is an empty serverlist, sent right away.
-  ScheduleResponseForBalancer(0, LoadBalanceResponse(), 0);
   // Send non-empty serverlist only after kServerlistDelayMs
   ScheduleResponseForBalancer(
-      0, BalancerServiceImpl::BuildResponseForBackends(GetBackendPorts(), {}),
+      0, 
+      BalancerServiceImpl::BuildResponseForBackends(
+          GetBackendPorts(kNumBackendInResolution /* begin */), {}),
       kServerlistDelayMs);
 
   const gpr_timespec second_call_time = gpr_time_add(
       gpr_now(GPR_CLOCK_REALTIME), 
       gpr_time_from_millis(kServerlistDelayMs * 1.1, GPR_TIMESPAN));
 
-  const auto t0 = system_clock::now();
   // The first request. The client will block while it's still trying to 
   // contact the balancer.
-  const auto& statuses_and_responses_1 =
-      SendRpc(kMessage_, num_backends_);
-  const auto ellapsed_ms_1 =
-      std::chrono::duration_cast<std::chrono::milliseconds>(
-          system_clock::now() - t0);
+  const auto& statuses_and_responses_1 = SendRpc(kMessage_, 
+                                                 kNumBackendInResolution);
 
-  // After fallback is in use, the call succeeds even though it's before 
-  // the serverlist is available.
-  EXPECT_GT(ellapsed_ms_1.count(), kFallbackTimeoutMs);
-  EXPECT_LT(ellapsed_ms_1.count(), kServerlistDelayMs);
-
-  // Each backend should have gotten 1 request.
+  // Fallback is used: each backend returned by the resolver should have 
+  // gotten one request.
   for (size_t i = 0; i < backends_.size(); ++i) {
```

Comment:
Suggest writing this as two separate loops, one from 0 to `kNumBackendInResolution - 1`, and the other from `kNumBackendInResolution` to `backends_.size()`.